subxt/src/custom_values/custom_value_address.rs

```diff
@@ -19,3 +27,33 @@ impl CustomValueAddress for str {
         self
     }
 }
+
+/// A static address to a custom value.
+pub struct StaticAddress<R> {
+    name: &'static str,
+    hash: [u8; 32],
+    phantom: PhantomData<R>,
+}
+
+impl<R> StaticAddress<R> {
+    /// Creates a new StaticAddress.
+    pub fn new(name: &'static str, hash: [u8; 32]) -> Self {
```

Comment:
For consistency with eg constants, can we call this `new_static` and make it `#[doc(hidden)]`. The reason for this is that "regular" users won't be able to provide a meaningful `hash` argument; only our static codegen can do that (and we reserve the right to change how we hash things, so allowing a user to save and provide hashes will just be a recipe for issues anyway :))

Let's also copy how constants is implemented by making the hash optional (ie `Option<[u8; 32]>`) and adding an `unvalidated()` function so that users of the static codegen have the option to opt out of the validation if they want to YOLO it.